src/mesh_analysis.rs

```diff
@@ -653,4 +720,79 @@ mod tests {
         );
         assert_eq!(genus, 3);
     }
+
+    #[test]
+    fn test_border_edge_loops_should_be_one() {
+        let (faces, vertices) = tessellated_triangle();
+        let geometry = Geometry::from_triangle_faces_with_vertices_and_computed_normals(
+            faces.clone(),
+            vertices.clone(),
+            NormalStrategy::Sharp,
+        );
+
+        let oriented_edges: Vec<OrientedEdge> = geometry.oriented_edges_iter().collect();
+        let edge_sharing_map = edge_sharing(&oriented_edges);
```

Comment:
Now I am wondering why `edge_sharing` takes oriented edges, only to convert them to unoriented internally. Let's not fix it now, but create a trello issue for later.

Reply:
there had to be some decision made. i suggest implementing the function for both, oriented and unoriented edge